lsp-mode.el

```diff
@@ -106,30 +148,27 @@ Optional arguments:
  a single argument (LSP workspace) and returning a plist is also accepted.
 
 `:initialize' is a function called when the client is intiailized. It takes a
- single argument, the newly created client."
-  (let ((enable (intern (format "%s-enable" name))))
-    `(defun ,enable ()
-       ,docstring
-       (interactive)
-       (let ((client (make-lsp--client
-                       :language-id (if ,language-id-fn ,language-id-fn #'(lambda (_) ,language-id))
-                       :send-sync #'lsp--stdio-send-sync
-                       :send-async #'lsp--stdio-send-async
-                       :new-connection (lsp--make-stdio-connection
+  single argument, the newly created client."
+  (let ((enable (intern (format "%s-enable-hook" name))))
```

Comment:
By renaming the entrypoint to `%s-enable-hook`, we'd be break `lsp-mode` for all people that currently use it in their configs. Is there a better way to achieve this?

Reply:
Hmm good question. We could increase the version number or maybe find some better trick :/